### PR TITLE
Allows globs for rules when unit testing 

### PR DESCRIPTION
**Pull request by keenanromain**
Solves #5507

Signed-off-by: Keenan Romain keenan.romain@gmail.com

### CODE REVIEW

**Comment by keenanromain** [comment]
@simonpasquier Sorry, I made a bit of a git mess in my last pull request so I created this new one and closed the last one. I placed the logic to handle the globs inside of the same function that resolves the filepaths. I also updated `docs/configuration/unit_testing_rules.md` and mentioned that globs are supported.

File: cmd/promtool/unittest.go [hunk 1]
```diff
@@ -128,14 +128,25 @@ func (utf *unitTestFile) maxEvalTime() time.Duration {
 	return maxd
 }
 
-// resolveFilepaths joins all relative paths in a configuration
-// with a given base directory.
-func resolveFilepaths(baseDir string, utf *unitTestFile) {
+// resolveAndGlobFilepaths joins all relative paths in a configuration
+// with a given base directory and replaces all globs with matching files.
+func resolveAndGlobFilepaths(baseDir string, utf *unitTestFile) {
```
Comment (simonpasquier) [inline cmd/promtool/unittest.go]: I would make `resolveAndGlobFilepaths` return an `error` and fail if something goes wrong with `filepath.Glob()`.

**Comment by keenanromain** [comment]
@simonpasquier How would we get this merged?

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**